### PR TITLE
DB-API instrument_connection accepts optional connect_module

### PR DESCRIPTION
**Pull request by tammy-baylis-swi**
# Description

DB-API `instrument_connection` accepts optional `connect_module`.

Should be merged before https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2942

Fixes https://github.com/open-telemetry/opentelemetry-python-contrib/issues/3028

Relates to https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2902

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Review by codefromthecrypt — APPROVED** [review]
Thanks, I suppose testing here was thought about but considered too much code for what it is covering?

**Review by aabmass — APPROVED** [review]

**Comment by aabmass** [comment]
> Thanks, I suppose testing here was thought about but considered too much code for what it is covering?

@tammy-baylis-swi any thoughts on this? I can merge once resolved

**Comment by tammy-baylis-swi** [comment]
> > Thanks, I suppose testing here was thought about but considered too much code for what it is covering?
> 
> @tammy-baylis-swi any thoughts on this? I can merge once resolved

Yeah more tests! I added some in [916bf5e](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3027/commits/916bf5e883cb296b9d3a9378a69cdc07bf3d753e). Let me know if any changes/more needed.

File: instrumentation/opentelemetry-instrumentation-dbapi/tests/test_dbapi_integration.py [hunk 1]
```diff
@@ -592,6 +592,49 @@ def test_instrument_connection(self):
...
+            tracer_provider=mock_tracer_provider,
+            capture_parameters=True,
+            enable_commenter=True,
+            commenter_options={"foo": "bar"},
+            connect_module=mock_connect_module,
+        )
+        kwargs = mock_dbapiint.call_args[1]
+        self.assertEqual(kwargs["connection_attributes"], {"foo": "bar"})
+        self.assertEqual(kwargs["version"], "test")
+        self.assertEqual(kwargs["tracer_provider"], mock_tracer_provider)
```
Comment (aabmass) [inline instrumentation/opentelemetry-instrumentation-dbapi/tests/test_dbapi_integration.py:632]: not sure how mocks behave with `==` check, you may want to do `assertIs()` to assert they are the same instance (and on L636)

File: instrumentation/opentelemetry-instrumentation-dbapi/tests/test_dbapi_integration.py [hunk 2]
```diff
@@ -592,6 +592,49 @@ def test_instrument_connection(self):
         connection2 = dbapi.instrument_connection(self.tracer, connection, "-")
         self.assertIs(connection2.__wrapped__, connection)
 
+    @mock.patch("opentelemetry.instrumentation.dbapi.DatabaseApiIntegration")
+    def test_instrument_connection_kwargs_defaults(self, mock_dbapiint):
+        connection = mock.Mock()
+        # Avoid get_attributes failing because can't concatenate mock
```
Comment (aabmass) [inline instrumentation/opentelemetry-instrumentation-dbapi/tests/test_dbapi_integration.py:598]: Not sure what this is referring to specifically, but you could probably achieve with MagicMock
```py
In [4]: mock.MagicMock() + []
Out[4]: <MagicMock name='mock.__add__()' id='140567852922320'>
```

**Review by aabmass — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-dbapi/tests/test_dbapi_integration.py (hunk 2 above)
Comment (tammy-baylis-swi) [reply instrumentation/opentelemetry-instrumentation-dbapi/tests/test_dbapi_integration.py]: I copied it from an existing test 😶 It seems not needed at all for the tests I've added, so removed in [ce86c1e](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3027/commits/ce86c1e3a1080ed574e136fe5cd525af072cc12b)

File: instrumentation/opentelemetry-instrumentation-dbapi/tests/test_dbapi_integration.py (hunk 1 above)
Comment (tammy-baylis-swi) [reply instrumentation/opentelemetry-instrumentation-dbapi/tests/test_dbapi_integration.py]: Yes! Changed in [ce86c1e](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3027/commits/ce86c1e3a1080ed574e136fe5cd525af072cc12b)

**Review by aabmass — APPROVED** [review]

**Pull request merged**